lib/api/controllers/adminController.js

```diff
@@ -93,10 +90,12 @@ class AdminController extends BaseController {
       cacheEngine = this.kuzzle.services.list[database];
 
     if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+      this.throw('database_not_found', database);
     }
 
-    cacheEngine.flushdb();
+    if (cacheEngine !== null && cacheEngine !== undefined) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "true". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2583)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "!==" check; it will always be true. Did you mean to use "!="? [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3403)